Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -122,34 +125,82 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
         guard let bvc = self.bvc else {
             return
         }
-        tableView.snp_updateConstraints { make in
-            make.bottom.equalTo(self.view).offset(bvc.shouldShowFooterForTraitCollection(newCollection) ? -UIConstants.ToolbarHeight : 0)
+        if bvc.shouldShowFooterForTraitCollection(newCollection) != snappedToBottom {
+            self.dismissing = true
```

Comment:
It was because `viewWillTransitionToSize` would override the height constraint set by `willTransitionToTraitCollection` — but now that I look at it, the animation is so fast compared to the dismissal that you can't notice a difference anyway (maybe this isn't actually an issue at all — it's hard to tell).